doc/radosgw: refine "bucket notifications"

zdover23 opened the pull request:
This commit refines the English in the "Bucket Notifications REST API" section of the notifications.rst page (the "Bucket Notifications" page).

Co-author: Cole Mitchell <cole.mitchell@gmail.com>

Signed-off-by: Zac Dover <zac.dover@gmail.com>





<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

colemitchell commented on doc/radosgw/notifications.rst:
```diff
@@ -161,55 +162,81 @@ name of an existing topic and different endpoint values).
+- push-endpoint: the URI of an endpoint to send push notifications to.
+- OpaqueData: opaque data is set in the topic configuration and added to all
+  notifications that are triggered by the topic.
+- persistent: indicates whether notifications to this endpoint are persistent
```
@zdover23 This and other _indicates whether_ entries (as well as line 165's push-endpoint entry) should perhaps be rewritten into complete sentences (if for no other reason than to match the OpaqueData entry and others).

colemitchell commented on doc/radosgw/notifications.rst:
```diff
@@ -161,55 +162,81 @@ name of an existing topic and different endpoint values).
- - port defaults to: 80/443 for HTTP/S accordingly
- - verify-ssl: indicate whether the server certificate is validated by the client or not ("true" by default)
- - cloudevents: indicate whether the HTTP header should contain attributes according to the `S3 CloudEvents Spec`_ ("false" by default)
+ - port: defaults to "80 for HTTP or 443 for HTTPS.
```
@zdover23 The quotation mark needs to be fixed.

colemitchell commented on doc/radosgw/notifications.rst:
```diff
@@ -161,55 +162,81 @@ name of an existing topic and different endpoint values).
+ - "user/password": defaults to "guest/guest".
+ - user/password: must be provided only over HTTPS. Topic creation requests will
+   otherwise be rejected.
+ - port: defaults to: 5672 for unencrypted connections and 5671 for
```
@zdover23 I think the second colon needs to be omitted.

colemitchell commented on doc/radosgw/notifications.rst:
```diff
@@ -161,55 +162,81 @@ name of an existing topic and different endpoint values).
+   otherwise be rejected.
+ - port: defaults to: 5672 for unencrypted connections and 5671 for
+   SSL-encrypted connections.
+ - vhost: defaults to: "/"
```
@zdover23 I think the second colon needs to be omitted.

colemitchell commented on doc/radosgw/notifications.rst:
```diff
@@ -161,55 +162,81 @@ name of an existing topic and different endpoint values).
+ - vhost: defaults to: "/"
+ - verify-ssl: indicates whether the server certificate is validated by the
+   client. (This is "true" by default.)
+ - if ``ca-location`` is provided and secure connection is used, the specified
```
@zdover23 I'm not sure this one should be lowercase like the others.

colemitchell commented on doc/radosgw/notifications.rst:
```diff
@@ -161,55 +162,81 @@ name of an existing topic and different endpoint values).
+ - if ``ca-location`` is provided and secure connection is used, the specified
+   CA will be used to authenticate the broker. The default CA will not be used. 
+ - amqp-exchange: the exchanges must exist and be able to route messages based
+   on topics. (This parameter is mandatory for AMQP0.9.1). Different topics
```
@zdover23 For consistency's sake, the period should be before the closing parenthesis.

zdover23 commented:
@colemitchell - I've addressed all your suggestions. I've also made a few refinements to the English on top of the refinements that were already there. I'll leave the conversations that you started open, and you can close them as you verify that the changes have been made.

colemitchell commented on doc/radosgw/notifications.rst:
```diff
@@ -161,55 +162,83 @@ name of an existing topic and different endpoint values).
-    - Attribute indexing does not need to be sequential or start from any specific value
-    - `AWS Create Topic`_ has a detailed explanation of the endpoint attributes format. However, in our case different keys and values are used
+    - The key-value pair of a specific parameter need not reside in the same
+      line as the parameter, and does not have to appear in any specific order,
```
@zdover23 For consistency's sake, I'd replace _does not have to appear_ with _need not appear_.

colemitchell commented on doc/radosgw/notifications.rst:
```diff
@@ -161,55 +162,83 @@ name of an existing topic and different endpoint values).
+    - Attribute indexing need not be sequential and need not start from any
+      specific value.
+    - `AWS Create Topic`_ provides a detailed explanation of the endpoint
+      attributes format. However, in our case different keys and values are
```
@zdover23 There needs to be another comma after _in our case_. Alternatively, it might read better as _In our case, however, different..._.

zdover23 commented:
As of the time of this comment (0722 aest 20 Oct 2022), the ceph API tests have been running for one hour.

zdover23 commented:
jenkins test api

zdover23 commented:
https://github.com/ceph/ceph/pull/48560 - Quincy backport
https://github.com/ceph/ceph/pull/48561 - Pacific backport
